main.js

```diff
@@ -1,15 +1,18 @@
 const { app } = require('electron')
+const { spawn } = require('child_process')
+
 const chokidar = require('chokidar')
 const fs = require('fs')
-const { spawn } = require('child_process')
+const path = require('path')
 
 const appPath = app.getAppPath()
 const ignoredPaths = /node_modules|[/\\]\./
+
 // Main file poses a special case, as its changes are
 // only effective when the process is restarted (hard reset)
 // We assume that electron-reload is required by the main
 // file of the electron application
-const mainFile = module.parent.filename
+const mainFile = path.normalize((module.parent || {}).filename || __filename)
```

Comment:
This defeats the purpose of hard resets as `__filename` is not the same as `module.parent.filename`.

Reply:
@yan-foto
So, I can add support only check this variable. Is it a suitable solution?

```javascript
 const mainFile = (module.parent || {}).filename || null

if (mainFile && eXecutable && fs.existsSync(eXecutable)) {
...
```

Reply:
There are two types of resets: hard and soft. `|| null` would render hard reset ineffective without knowing why it happened. It might solve your specific problem but would weaken the logic of the program.

I though you have found a solution that does not require changing `electron-reload`. 🤔